crates/bevy_core_pipeline/src/lib.rs

```diff
@@ -243,10 +264,34 @@ impl BatchedPhaseItem for Transparent2d {
 }
 
 pub struct Opaque3d {
-    pub distance: f32,
-    pub pipeline: CachedRenderPipelineId,
-    pub entity: Entity,
-    pub draw_function: DrawFunctionId,
+    distance: f32,
+    pipeline: CachedRenderPipelineId,
+    entity: Entity,
+    draw_function: DrawFunctionId,
+}
+
+impl Opaque3d {
+    #[inline]
+    pub fn from_mesh_transform(
```

Comment:
Would there ever be a use case for creating an `Opaque3d` without first calculating the matrices? It looks like `from_mesh_transform` is the only way to construct the phase item now, which may be restricting?

Could be solved by either making the field public again or adding another unopinionated constructor if this is even useful.